lib/hacker_news/vote_lookup_service.rb

```diff
@@ -0,0 +1,19 @@
+module HackerNews
+  class VoteLookupService
+
+    attr_reader :user, :votable
+    def initialize(user, votable)
+      @user = user
+      @votable = votable
+    end
+
+    def vote
+      if @vote ||= Vote.find_by(user_id: user.id, votable: votable, votable_type: votable.class.to_s)
+        @vote
+      else
+        nil
+      end
```

Comment:
You don't need the `if/else` here.  Doing `@vote ||=` is going to net the exact same behavior.